pkg/build/buildapihelpers/util.go

```diff
@@ -29,26 +30,6 @@ func StrategyType(strategy BuildStrategy) string {
 	return ""
 }
 
-func SourceType(source BuildSource) string {
-	var sourceType string
-	if source.Git != nil {
-		sourceType = "Git"
-	}
-	if source.Dockerfile != nil {
-		if len(sourceType) != 0 {
-			sourceType = sourceType + ","
-		}
-		sourceType = sourceType + "Dockerfile"
-	}
-	if source.Binary != nil {
-		if len(sourceType) != 0 {
-			sourceType = sourceType + ","
-		}
-		sourceType = sourceType + "Binary"
-	}
-	return sourceType
-}
-
 // LabelValue returns a string to use as a value for the Build
```

Comment:
This method is highly questionable.  It's being used on an already compressed name and if the field it's setting cannot be set it would actually be better to fail since we really don't want to compress that value-space.

Reply:
it's not an already compressed name in the case that we've taken the name and appended a sequence number to it, which i'm pretty sure is how we ended up needing this.

(someone has a buildconfig w/ a maxed out name.  Without this logic we then append "-1" to it and fail to create the label).